### DISCUSSION TITLE
Angular tsParticles Component

### DISCUSSION THREAD

**Discussion by matteobruni** [matteobruni/tsparticles#1126] [General]
This section will be used to talk about the `Angular` version of tsParticles, the `ng-particles` package.

**Comment by LeSleedz** [comment]
Hi matteo, 
First of all thanks for all the work you've put in this.
I tried ng particles in angular but it generates a few errors.
Here is a screen of what happens after ng serve.
What's weird is that sometimes despites these errors it will sometime still compiles and i will be able to see the animations.
![ngParticles errors](https://user-images.githubusercontent.com/17353624/103333209-1a516580-4a3b-11eb-903c-11954caa2a79.png)

**Reply by LeSleedz** [reply]
I found a quick fix but it's not great tbh : since the compiler was seeing 'out' as a string i casted it as 'out' . Same for a few other attributes such as move.direction for example
![fix](https://user-images.githubusercontent.com/17353624/103334232-a3b66700-4a3e-11eb-924f-bd2733a3130c.png)

**Reply by matteobruni** [reply]
Try using the enums instead, they should work better.

You can see a sample [here](https://github.com/matteobruni/tsparticles/blob/4340750b22588bc764845f58b2848377b20ca9ee/demo/react/src/frames.ts#L78)

I’ll investigate the string issue anyway, it’s really weird

**Reply by MaEdAlpha** [reply]
I too have a similar issue: When hovering over the [options] field I get this message.

` The types of 'particles.move.outMode' are incompatible between these types.
    Type 'string' is not assignable to type '"split" | "out" | "bounce" | "destroy" | OutMode | "bounceHorizontal" | "bounceVertical" | "none" | OutModeAlt | undefined'.`
![tsparticlebug](https://user-images.githubusercontent.com/72553021/141762663-7086f2cd-449e-46a7-adbe-19ab904e4323.PNG)

LeSleedz's hack worked. Thank you very much!

**Comment by craftsolution** [comment]
Hello,

Thanks for your works.

I have integrated tsParticles on my angular project but I have some questions:

- Is it possible to add text hover it ? 
- Or is it possible to ad it on div ? 
-  How to remove the scroll bar that I have ? 

Best regards,

**Reply by craftsolution** [reply]
![image](https://user-images.githubusercontent.com/50966314/126041690-fe9bee92-4642-43d1-b186-fc07864e9f05.png)

**Reply by matteobruni** [reply]
Hi @MainTendue, the particles animation is a standard HTML element so you can use CSS to give it the size you want, put it in the back or reduce its size.

About the scrollbar I don't know what is the issue, maybe you made some mistakes with the CSS, the `fullScreen` options sets it without any scrollbar and I'm 100% sure about it.

**Reply by craftsolution** [reply]
<ng-particles [id]="id" [options]="particlesOptions" (particlesLoaded)="particlesLoaded($event)" (particlesInit)="particlesInit($event)">
</ng-particles>

So I can use for exemple in scss #id or ng-particles ? I am using Angular.

**Reply by craftsolution** [reply]
Here is what i idid

ng-particles {
    max-width: 100%;
    height: 100vw;
    overflow: hidden;
    margin: 0;
    padding: 0;

}

is it correct ?

**Reply by matteobruni** [reply]
If you want particles to be full screen remove the CSS and just add `fullScreen: { enable: true }` to your options. I never recommend to use CSS for full screen since it's already wrapped and tested in the library.

**Reply by craftsolution** [reply]
Thanks. it works. for fullscreen. 

for the html element you said I can you ng-particles as I used in css ?

**Reply by matteobruni** [reply]
Yes, or you can use the id assigned to properties

**Reply by craftsolution** [reply]
Thanks for your help

**Reply by craftsolution** [reply]
Hello, 

I am having an error while building the angular app : 

ERROR Private identifiers are only available when targeting ECMAScript 2015 and higher.

![image](https://user-images.githubusercontent.com/50966314/126068420-946746a0-3fc8-4095-8204-0f058b3ef23f.png)

**Reply by craftsolution** [reply]
Do you have an idea ? 
 I have tried to comment it this way  but no chance

![image](https://user-images.githubusercontent.com/50966314/126068449-fa74516a-6cdf-4fd0-944b-483471c7904c.png)

**Reply by matteobruni** [reply]
ECMAScript 5 and below are not supported, you have to build for ES6 (or ES2015) or newer

**Reply by craftsolution** [reply]
in tsconfig.json file I have es2015 which give me error

![image](https://user-images.githubusercontent.com/50966314/126069253-ca4ed7c6-3fb3-467d-a739-e0409d008dee.png)

**Reply by matteobruni** [reply]
Maybe some settings in the `angular.json` file?

**Reply by craftsolution** [reply]
Nope. it is in the tsconfig.js 

I have found the correct value to use

![image](https://user-images.githubusercontent.com/50966314/126073744-28b07f2c-ea35-47f6-82f8-3ad1d4e62a02.png)

it may help someone else.
thanks

**Comment by matteobruni** [comment]
This discussion will be closed since there's a dedicated repository for the Angular component: https://github.com/tsparticles/angular
